src/components/AvgMonthPaymentWidget/AvgMonthPaymentWidget.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Crm\PaymentsModule\Components;
+
+use Crm\ApplicationModule\Widget\BaseWidget;
+use Crm\ApplicationModule\Widget\WidgetManager;
+use Nette\Database\Connection;
+
+class AvgMonthPaymentWidget extends BaseWidget
+{
+    private $templateName = 'avg_month_payment_widget.latte';
+
+    private $database;
+
+    public function __construct(WidgetManager $widgetManager, Connection $database)
+    {
+        parent::__construct($widgetManager);
+        $this->database = $database;
+    }
+
+    public function identifier()
+    {
+        return 'avgmonthpaymentwidget';
+    }
+
+    public function render(array $params)
+    {
+        $segment = $params['segment'];
+        $data = $params['data'];
+        $usersIds = [];
+        $tableData = [];
+        $displayFields = false;
+
+        $segment->process(function ($row) use (&$usersIds, $data, &$tableData, &$displayFields) {
+            $usersIds[] = $row->id;
+
+            if ($data) {
+                if (!$displayFields) {
+                    $displayFields = array_keys((array) $row);
+                }
+                $tableData[] = array_values((array) $row);
+            }
+        }, 100000);
+
+        $this->template->fields = $displayFields;
+        $this->template->data = $tableData;
+
+        if (count($usersIds)) {
+            $usersIds = implode(',', $usersIds);
+            $query = "SELECT AVG(value) AS avg_month_payment FROM user_meta WHERE `key`='avg_month_payment' AND user_id IN (".$usersIds.")";
```

Comment:
You can get rid of the raw query and write "nice" Nette Database query instead.

Reply:
Can I use UserMetaRepository? It is in crm-users-module but I suppose that without that module this widget us useless.

Reply:
You can use the repository. `PaymentsModule` -> `UsersModule` dependency is fine, as each payment needs to belong to some user anyway.